app/utils/crypto/symmetric/recrypt.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+class Crypto::Symmetric::Recrypt
+
+  def initialize(current_user, team, private_key)
+    @current_user = current_user
+    @team = team
+    @private_key = private_key
+  end
+
+  def perform
+    return if latest_in_use? || recrypt_not_ready?
```

Comment:
maybe find a better name?
latest? what does latest mean here?
**already_recrypted?**

or .... requires_recrypt? and invert with unless